pkg/process/manager.go

```diff
@@ -234,10 +241,14 @@ func (m *Manager) DestroyDaemon(d *daemon.Daemon) error {
 		if err = p.Signal(syscall.SIGTERM); err != nil {
 			return errors.Wrapf(err, "send SIGTERM signal to process %d", d.Pid)
 		}
-		// if nydus-snapshotter restart, it will break the relationship between nydusd and
+
+		// if nydus-snapshotter restarts, it will break the relationship between nydusd and
 		// nydus-snapshotter, p.Wait() will return err, so here should exclude this case
-		if _, err = p.Wait(); err != nil && !stderrors.Is(err, syscall.ECHILD) {
-			log.L.Errorf("failed to process wait, %v", err)
+		if state, err := p.Wait(); err == nil {
+			// Expect to catch failure exit status.
+			d.SuccessStatus = state.Success()
+		} else if err != nil && !stderrors.Is(err, syscall.ECHILD) {
```

Comment:
Maybe we can give a warning for `ECHILD` status.

Reply:
Generally, it is a normal situation. But I can add a warning log entry there.

Reply:
This code snippet is no longer touched.